apps/financial-aid/api/src/app/modules/auth/auth.controller.ts

```diff
@@ -172,7 +174,17 @@ export class AuthController {
 
     const jwtToken = this.sharedAuthService.signJwt(user, csrfToken)
 
-    const returnUrl = user.service === 'osk' ? '/umsokn' : '/nymal'
+    if (res.req?.query.service === 'veita') {
+      user.returnUrl = ReturnUrl.VEITA
+    }
+    if (res.req?.query.applicationId) {
+      user.returnUrl = ReturnUrl.STADA
+    }
+
+    const returnURl =
+      user.returnUrl === ReturnUrl.STADA
+        ? ReturnUrl.STADA + '/' + res.req?.query.applicationId
+        : user.returnUrl
```

Comment:
You could add the applicationId param on line 181 and then you don't have to have this check here.

Reply:
true, but I cant set the user.returnurl with application id it only accepts enum, so I will always have to add another variable anyways?

Reply:
So what I am thinking is basically you could drop lines 177-187 and instead do something like: 
```
        const returnUrl = `${user.returnUrl}${
          res.req?.query.applicationId ? `/${res.req.query.applicationId}` : ''
        }`
```
Or you could even do it in the redirect on line 198 or am I misunderstanding something? 👼